transformer_engine/jax/cpp_extensions/custom_call.py

```diff
@@ -93,6 +102,40 @@ def custom_caller(name, args, opaque, has_side_effect, **kwargs):
             result_layouts=args.output_layouts,
             backend_config=opaque,
             has_side_effect=has_side_effect,
-            **kwargs
+            **kwargs,
         )
     return out
+
+
+#   Reference for custom_call from JAX
+#   (https://github.com/google/jax/blob/956226c929a34741c1f0a396b96e721c61ae5794/
+#   jax/_src/interpreters/mlir.py#L2589)
+#    def custom_call(
+#        call_target_name: str,
+#        *,
+#        result_types: Sequence[ir.Type],
+#        operands: Sequence[ir.Value],
+#        backend_config: str | bytes | dict[str, ir.Attribute] = "",
+#        has_side_effect: bool = False,
+#        result_shapes: Sequence[ir.Value] | None = None,
+#        called_computations: Sequence[str] = (),
+#        api_version: int = 2,
+#        operand_output_aliases: dict[int, int] | None = None,
+#        operand_layouts: Sequence[Sequence[int]] | None = None,
+#        result_layouts: Sequence[Sequence[int]] | None = None,
+#        extra_attributes: dict[str, ir.Attribute] | None = None,
+#        ) -> ir.Operation:
+
+
+def custom_caller_with_ffi(name, args, backend_config, **kwargs):
+    """
+    New XLA custom call warpper
+    """
+    return mlir.custom_call(
+        call_target_name=name,
+        result_types=args.output_types,
+        operands=args.operands,
+        backend_config=backend_config,
+        api_version=4,
```

Comment:
What does 4 mean in this context?

Reply:
`api_version=2` for the old custom call and `4` for the new one ([example](https://github.com/google/jax/blob/988ff27dfa03bb2102efefb48885ffe582be834a/jaxlib/gpu_prng.py#L123)).
I first implemented this`custom_caller_with_ffi()` when the `ffi.ffi_lowering()` was not yet there. 
Since the `ffi.ffi_lowering` seems to work fine for our needs now, I'm considering to remove this function.